### DISCUSSION TITLE
Error message

### DISCUSSION THREAD

**Discussion by 64Zone** [EpiModel/EpiModel#620] [EpidemicMethods] [answered]
Happy new year!

I was fitting a base model (i.e., Example 1: Independent SIS Model) in the R tutorial. However, running the diagnostic line always generate the following error:

> dx <- netdx(est1, nsims = 10, nsteps = 1000) 
> Error in get(name, pos = environment(new)) : 
>   argument "set.control.stergm" is missing, with no default

This line used to work for me and I am not sure what are the causes. I was using 1.6.5 EpiModel package and here is my sessionInfo() output:

> R version 4.1.1 (2021-08-10)
> Platform: x86_64-apple-darwin17.0 (64-bit)
> Running under: macOS Big Sur 11.6
> 
> Matrix products: default
> LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
> 
> locale:
> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
>  [1] devtools_2.4.2        usethis_2.0.1         tictoc_1.0.1          EpiModel_1.6.5        tergm_4.0.2           ergm_4.1.2           
>  [7] networkDynamic_0.11.0 network_1.17.1        deSolve_1.28          janitor_2.1.0         forcats_0.5.1         stringr_1.4.0        
> [13] dplyr_1.0.7           purrr_0.3.4           readr_2.0.1           tidyr_1.1.3           tibble_3.1.3          ggplot2_3.3.5        
> [19] tidyverse_1.3.1      
> 
> loaded via a namespace (and not attached):
>  [1] nlme_3.1-152            fs_1.5.0                lubridate_1.7.10        doParallel_1.0.16       RColorBrewer_1.1-2      httr_1.4.2             
>  [7] rprojroot_2.0.2         tools_4.1.1             backports_1.2.1         utf8_1.2.2              R6_2.5.0                DBI_1.1.1              
> [13] lazyeval_0.2.2          colorspace_2.0-2        withr_2.4.2             prettyunits_1.1.1       processx_3.5.2          tidyselect_1.1.1       
> [19] compiler_4.1.1          cli_3.0.1               rvest_1.0.1             xml2_1.3.2              desc_1.3.0              scales_1.1.1           
> [25] DEoptimR_1.0-9          robustbase_0.93-8       callr_3.7.0             digest_0.6.27           rmarkdown_2.10          pkgconfig_2.0.3        
> [31] htmltools_0.5.1.1       sessioninfo_1.1.1       dbplyr_2.1.1            fastmap_1.1.0           rlang_0.4.11            readxl_1.3.1           
> [37] rstudioapi_0.13         generics_0.1.0          jsonlite_1.7.2          statnet.common_4.5.0    magrittr_2.0.1          Matrix_1.3-4           
> [43] Rcpp_1.0.7              munsell_0.5.0           fansi_0.5.0             ape_5.5                 lifecycle_1.0.0         stringi_1.7.3          
> [49] yaml_2.2.1              snakecase_0.11.0        MASS_7.3-54             pkgbuild_1.2.0          grid_4.1.1              parallel_4.1.1         
> [55] crayon_1.4.1            lattice_0.20-44         haven_2.4.3             lpSolveAPI_5.5.2.0-17.7 hms_1.1.0               ps_1.6.0               
> [61] knitr_1.33              pillar_1.6.2            pkgload_1.2.1           codetools_0.2-18        rle_0.9.2               reprex_2.0.1           
> [67] glue_1.4.2              evaluate_0.14           trust_0.1-8             remotes_2.4.0           modelr_0.1.8            vctrs_0.3.8            
> [73] tzdb_0.1.2              foreach_1.5.1           testthat_3.0.4          cellranger_1.1.0        gtable_0.3.0            assertthat_0.2.1       
> [79] cachem_1.0.5            xfun_0.25               broom_0.7.9             coda_0.19-4             iterators_1.0.13        memoise_2.0.0          
> [85] ellipsis_0.3.2 

By the way, I wonder if there is an updated version of R tutorial that corresponds to the current version of EpiModel? Thank you very much!!

- Sicong

**Comment by sgoodreau** [comment]
Hi Sicong - 

Thanks for all the detail, especially the session info.  It looks like you have very new versions of the `statnet` packages (`ergm`, `tergm`) but a rather old version of `EpiModel` itself.  We're pretty confident that is creating the conflict you see, as there were some changes involving `set.control.stergm` that we made that invoved all three pacakges.  Try updating `EpiModel` to the latest version (v2.2.0) and see if that solves it.

Best,
Steve

**Reply by 64Zone** [reply]
Thank you for your fast reply Steve! 

I'd be very happy to update EpiModel package. I wonder if you know any updated R tutorial files? I noticed frequent syntax changes through the versions of EpiModel and it will be very helpful for me. Thank you!!

Cheers,
Sicong

**Comment by martinamorris** [comment, marked as answer]
Hi @64Zone -- There were some major updates to EpiModel as the version changed to 2.0.  There's some info and a link to a migration vignette on the [EpiModel landing page](http://www.epimodel.org/)

**Reply by 64Zone** [reply]
Thank you @martinamorris ! I am trying to learn and apply EpiModel to HIV research. EpiModel is really interesting work and I look forward to learning from the team and am certain that I have more questions in store. Thank you all for the fast and helpful responses!
